docs/source/reference/ha.rst

```diff
@@ -0,0 +1,345 @@
+
+
+2. Fix listen address in `/etc/postgresql/9.3/main/postgresql.conf` and have PostgreSQL listen an interface that has an
+   IP address reachable for ``st2-multi-node-1`` and ``st2-multi-node-2``.
+
+3. To `/etc/postgresql/9.3/main/pg_hba.conf` add ACL rule. Here the subnet to allow access is `10.0.3.1/24`.3
+
+  .. code-block:: bash
+
+        host       all  all  10.0.3.1/24  trust
+
+4. restart PostgreSQL
+
+  .. code-block:: bash
+
+         service postgresql restart
+
+5. Create Mistral DB in PostgreSQL.
+
+  .. code-block:: bash
+
+        cat << EHD | sudo -u postgres psql
+        CREATE ROLE mistral WITH CREATEDB LOGIN ENCRYPTED PASSWORD 'StackStorm';
+        CREATE DATABASE mistral OWNER mistral;
+        EHD
+
+6. Add stable |st2| repos.
+
+  .. code-block:: bash
+
+        curl -s https://packagecloud.io/install/repositories/StackStorm/staging-stable/script.deb.sh | sudo bash
+
+7. Setup st2web and SSL termination. Follow :ref:`install webui and setup ssl<ref-install-webui-ssl-deb>`.
+
+8. Configuration for Nginx as loadbalancer for controller box can be found
+`here <https://gist.github.com/manasdk/fce14029900e533a385d#file-shared_st2_Nginx-conf>`__. With this configuration
+Nginx will loadbalance all requests between the two blueprint boxes ``st2-multi-node-1`` and ``st2-multi-node-2``.
+This includes requests to ``st2api``, ``st2auth`` and ``mistral-api``. Nginx also serves as the webserver for st2web.
+
+9. Install st2chatops following from :ref:`setup chatops<ref-setup-chatops-deb>`.
+
+Blueprint box
+^^^^^^^^^^^^^
+This box is a repeatable |st2| image that is essentially the single-box reference deployment with a few changes. The aim is
+to Deploy as many of these boxes for desired HA objectives and also get some horizontal scaling. |st2| processes outlined
+above support the capbility of being turned on-off individually therefore each box can also be made to offer different services.
+
+1.  Add stable |st2| repos.
+
+  .. code-block:: bash
+
+        curl -s https://packagecloud.io/install/repositories/StackStorm/staging-stable/script.deb.sh | sudo bash
+
+2. Install all |st2| components and mistral.
+
+  .. code-block:: bash
+
+        `sudo apt-get install -y st2 st2mistral`
+
+3. Update Mistral connection to PostgreSQL in `/etc/mistral/mistral.conf` by changing `database.connection` property.
+
+4. Update Mistral connection to RabbitMQ in `/etc/mistral/mistral.conf` by changing `default.transport_url` property.
+
+4. Setup users as per :ref:`here<ref-config-ssh-sudo-deb>`. Make sure that user configuration on all boxes running ``st2auth`` is
+   identical. This ensures consistent authentication from the entire |st2| install since the request to authenticate a user
+   can be forwarded by the loadbalancer to any of the ``st2auth`` processes.
+
+5. Use `shared st2 config <https://gist.github.com/manasdk/fce14029900e533a385d#file-st2-conf>`__ and replace `/etc/st2/st2.conf`.
```

Comment:
Yeah for centralized logging. The log strategy to use is totally up to you though - can leave logs on every box or even ship to logstash, loggly etc. I just picked one I liked for the purpose of this doc.